### ISSUE TITLE
Skills being parsed with the same name and path as last installed skill

### ISSUE THREAD

**Issue by FabioRosado** [opsdroid/opsdroid#227]
> Opsdroid.Skills
>
The key "config" from the skills dictionary is getting the name, path, module_path, install patch and other attributes (such as no-cache: True) from the last installed skill. 
This happens for every single skill, so when we do **help skills** the only skill shown is the latest installed skill.

```python
[{'regex': {'expression': 'hi|hello|hey|hallo', 'case_sensitive': True}, 'skill': <function hello at 0x1107cebf8>, 'config': {'name': 'test', 'path': '/Users/fabiorosado/Documents/GitHub/skill-test', 'type': 'skill', 'module_path': 'opsdroid-modules.skill.test', 'install_path': '/Users/fabiorosado/.opsdroid/modules/opsdroid-modules/skill/test', 'branch': 'master'}}, 
{'regex': {'expression': "bye( bye)?|see y(a|ou)|au revoir|gtg|I(\\')?m off", 'case_sensitive': True}, 'skill': <function goodbye at 0x1107ce730>, 'config': {'name': 'test', 'path': '/Users/fabiorosado/Documents/GitHub/skill-test', 'type': 'skill', 'module_path': 'opsdroid-modules.skill.test', 'install_path': '/Users/fabiorosado/.opsdroid/modules/opsdroid-modules/skill/test', 'branch': 'master'}}, 
{'regex': {'expression': '.*dance.*', 'case_sensitive': True}, 'skill': <function dance at 0x1107ceae8>, 'config': {'name': 'test', 'path': '/Users/fabiorosado/Documents/GitHub/skill-test', 'type': 'skill', 'module_path': 'opsdroid-modules.skill.test', 'install_path': '/Users/fabiorosado/.opsdroid/modules/opsdroid-modules/skill/test', 'branch': 'master'}}, 
{'regex': {'expression': '^help skill(.*?)$', 'case_sensitive': True}, 'skill': <function help_skill at 0x1108090d0>, 'config': {'name': 'test', 'path': '/Users/fabiorosado/Documents/GitHub/skill-test', 'type': 'skill', 'module_path': 'opsdroid-modules.skill.test', 'install_path': '/Users/fabiorosado/.opsdroid/modules/opsdroid-modules/skill/test', 'branch': 'master'}}, 
{'regex': {'expression': 'test', 'case_sensitive': True}, 'skill': <function test at 0x1107cef28>, 'config': {'name': 'test', 'path': '/Users/fabiorosado/Documents/GitHub/skill-test', 'type': 'skill', 'module_path': 'opsdroid-modules.skill.test', 'install_path': '/Users/fabiorosado/.opsdroid/modules/opsdroid-modules/skill/test', 'branch': 'master'}}]
```

### Tests Done:

- Noticed devtools **help skills** getting wrong value on my mac after opsdroid update
- Installed opsdroid on a linux machine, tested devtools **help skills** first - (no change compared to mac)
- Created test skill to see if the config values would change - (they did, all of them inherited from test)
- Tested opsdroid on Slack - (test values still showing)
- Tested opsdroid desktop - (test values still showing)

### Checks done on the code:
Method: used _LOGGER.info to print opsdroid.skills list
* core.py
  * function: setup_skills - (it shows the different names and values of each skill)
  * function: start_loop - (it shows the different names and values of each skill)
* loader.py
  * function: load_modules_from_config - (it shows the different names and values of each skill)
  * function: _load_modules - (it shows the different names and values of each skill)
* web.py
  * function: start - (it shows the test value on all skills)
* Others:
  * Tried to remove the no-cache command from the configuration - it made no difference

At the moment it's all I've done to try and tackle this issue. I'm unsure where the issue might be but I'll keep trying on my free time to see if I can figure out why the opsdroid.skills are being parsed with the wrong values in the config key.

**Comment by jacobtomlinson** [comment]
This could be an assignment issue where a set of pointers are being overwritten.

Could you try changing [this line](https://github.com/opsdroid/opsdroid/blob/master/opsdroid/loader.py#L236) from

```
"config": config})
```

to

```
"config": copy.copy(config)})
```

and also adding this import to the file

```
import copy
```

**Comment by jacobtomlinson** [comment]
Ok I've tried that and it didn't work. I think the issue is around the way that matchers pair the config with the functions.

You are seeing differences because in some places you are logging the skill modules, and in others you're logging the skill functions. Frustratingly they are both lists names `skills` so I see why you did that.

In the `core.py` in `start_loop` if you log `skills` and `self.skills` you'll see what I mean.

**Comment by jacobtomlinson** [comment]
If you look [here](https://github.com/opsdroid/opsdroid/blob/master/opsdroid/loader.py#L231) you can see that `loader.current_import_config` is being set to the config of the current module being imported. Then it calls `import_module` which should refer back to the `current_import_config` for the value. However it must not be running in the order I imagined.

**Comment by jacobtomlinson** [comment]
Changing `_reload_module` in `loader.py` to look something like this should fix the issue.

```python
    def _reload_modules(self, modules):
        for module in modules:
            self.current_import_config = module["config"]
            importlib.reload(module["module”])
```

**Comment by jacobtomlinson** [comment]
This bug caused skills to get the wrong config and therefore not function correctly. This has now been resolved in #228.
